File: config/config_manager.py
import json
import os

CONFIG_FILE = "config.json"

def salvar_config(config):

    with open(CONFIG_FILE, "w") as f:
        json.dump(config, f, indent=4)

def carregar_config():

    if not os.path.exists(CONFIG_FILE):
        return None

    with open(CONFIG_FILE) as f:
        return json.load(f)

def config_existe():
    return os.path.exists(CONFIG_FILE)

File: AgenteFiscal/AgenteFiscal/Utils/ConfigManager.cs
using System;
using System.IO;
using Newtonsoft.Json;

namespace AgenteFiscal.Utils
{
    public class Config
    {
        public string CNPJ { get; set; }
        public string Token { get; set; }
        public string ApiUrl { get; set; }
        public string XmlPath { get; set; }
    }

    public static class ConfigManager
    {
        static string basePath = Path.Combine(
            Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
            "AgenteFiscal"
        );

        static string configPath = Path.Combine(basePath, "config.json");

        public static bool Exists()
        {
            return File.Exists(configPath);
        }

        public static Config Load()
        {
            var json = File.ReadAllText(configPath);
            return JsonConvert.DeserializeObject<Config>(json);
        }

        public static void Save(Config config)
        {
            Directory.CreateDirectory(basePath);

            var json = JsonConvert.SerializeObject(config, Formatting.Indented);

            File.WriteAllText(configPath, json);
        }
    }
}